### PR TITLE
docs: add notice of early response in hooks (#3135)

### PR DESCRIPTION
**Pull request by seia-soto**
#### Summary

This PR is for #3135. Also, feedbacks to edit or improve this new line are all welcome.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by zekth — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by seia-soto** [comment]
Thank you all.

File: docs/Routes.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ They need to be in
 
   `reply` is defined in [Reply](Reply.md).
 
+**Notice:** The documentation of `onRequest`, `preParsing`, `preValidation`, `preHandler`, `preSerialization`, `onSend`, and `onResponse` are described in more details in [Hooks](Hooks.md) and to send response before request handled by `handler`, you need to refer [Respond to a request from a hook](Hooks.md#respond-to-a-request-from-a-hook) section for best practices.
```
Comment (jsumners) [inline docs/Routes.md:72]:
```suggestion
**Notice:** The documentation of `onRequest`, `preParsing`, `preValidation`,
`preHandler`, `preSerialization`, `onSend`, and `onResponse` are described in
more detail in [Hooks](Hooks.md). Additionally, to send a response before the
request is handled by the `handler` please refer to
[Respond to a request from a hook](Hooks.md#respond-to-a-request-from-a-hook).
```

**Pull request merged**